sequencer/src/api.rs

```diff
@@ -318,23 +317,10 @@ impl<
                 // map to ensure consistency.
                 let mut state = ValidatedState::from_header(leaf.block_header());
                 state.fee_merkle_tree = tree.clone();
-                let res = (Arc::new(state), None, Committable::commit(&leaf));
-                consensus
-                    .update_saved_leaves(leaf, &handle.hotshot.upgrade_lock)
-                    .await;
-                res
+                (Arc::new(state), None)
             }
         };
-        if let Err(err) = consensus.update_validated_state_map(
-            view,
-            View {
-                view_inner: ViewInner::Leaf {
-                    state,
-                    delta,
-                    leaf: leaf_commit,
-                },
-            },
-        ) {
+        if let Err(err) = consensus.update_leaf(leaf, Arc::clone(&state), delta) {
```

Comment:
`update_saved_leaves` and `update_validated_state_map` are now private. I figured `update_leaf` should cover the need for both of those